tests/api/Jalasoft.TeamUp.Projects.API.Tests/PostProjectsTest.cs

```diff
@@ -0,0 +1,34 @@
+namespace Jalasoft.TeamUp.Projects.API.Tests
+{
+    using Jalasoft.TeamUp.Projects.API.Controllers;
+    using Jalasoft.TeamUp.Projects.Core.Interfaces;
+    using Jalasoft.TeamUp.Projects.Models;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Moq;
+    using Xunit;
+
+    public class PostProjectsTest
+    {
+        private readonly Mock<IProjectsService> mockProjectsService;
+        private readonly DefaultHttpContext mockHttpContext;
+        private readonly PostProject postProject;
+
+        public PostProjectsTest()
+        {
+            this.mockProjectsService = new Mock<IProjectsService>();
+            this.mockHttpContext = new DefaultHttpContext();
+            this.postProject = new PostProject(this.mockProjectsService.Object);
+        }
+
+        [Fact]
+        public async void PostProject_Returns_OkObjectResult_Project()
+        {
+            var request = this.mockHttpContext.Request;
+            this.mockProjectsService.Setup(service => service.PostProject(null)).Returns(new Project());
```

Comment:
please instead of null, could you set up something like this to setup the body
public static Stream SetStream(string body)
        {
            var stream = new MemoryStream();
            var writer = new StreamWriter(stream);
            writer.Write(body);
            writer.Flush();
            stream.Position = 0;
            return stream;
        }

that will be called in your test

HttpRequest whatever = ...;
            whatever.Headers.Add("somename", "somevalue");
            whatever.Body = SetStream(
@"{ ""mybodyproperty"": ""myvalueproperty""}"
)